similar/main/game.cpp

```diff
@@ -1307,12 +1307,15 @@ window_event_result game_handler(window *,const d_event &event, const unused_win
 			Game_wind = NULL;
 			event_toggle_focus(0);
 			key_toggle_repeat(1);
+			return window_event_result::ignored;
 			break;
 
 		default:
 			break;
 	}
-	return window_event_result::ignored;
+	
+	// If we deleted the window, tell the event loop
+	return Game_wind ? window_event_result::ignored : window_event_result::deleted;
```

Comment:
OK, for example there's a call to `window_close(Game_wind)` in `GameProcessFrame`, which occurs when a demo has finished playing. Within that call to `window_close(Game_wind)` it calls `game_handler` to see how to handle `window_close` (via `EVENT_WINDOW_CLOSE`), which simply does some tidying up and sets `Game_wind` to `NULL`, then returns `window_event_result::ignored`. This tells `window_close` it may delete the window itself.

Now comes the tricky part. After it has finished the call to `window_close`, it's still in `GameProcessFrame`. `GameProcessFrame` promptly returns, but it's still responding to an `EVENT_WINDOW_DRAW` in `game_handler`. If we return `event_window_result::ignored` in response to this `EVENT_WINDOW_DRAW`, then `window_send_event` will faithfully return the same to `event_process`, which will then go on to call `window_get_next` on a dangling `wind` pointer (since `Game_wind` has been freed within the handler), which **may** result in a crash (doesn't seem to on my system).